hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/functional/TestHoodieBackedMetadata.java

```diff
@@ -1466,6 +1466,7 @@ private void verifyMetadataMergedRecords(HoodieTableMetaClient metadataMetaClien
         .withRequestedSchema(schema)
         .withDataSchema(schema)
         .withProps(new TypedProperties())
+        .withEnableOptimizedLogBlockScan(writeConfig.enableOptimizedLogBlocksScan())
```

Comment:
This needs to be metadata config since we are reading the MDT records here

Reply:
will fix this